.github/workflows/docker-build.yml

```diff
@@ -33,18 +33,37 @@ jobs:
 
       # Action reference: https://github.com/docker/login-action
       - name: Login to GitHub Container Registry
-        uses: docker/login-action@v1 
+        uses: docker/login-action@v1
         with:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
- 
+
       # Action reference: https://github.com/docker/build-push-action
-      - name: Build container
+      - name: Build base container
         uses: docker/build-push-action@v2
         with:
           context: ./deploy/base
           platforms: linux/amd64,linux/arm64,linux/ppc64le,linux/s390x
           push: true
           # keep tag in sync with deploy/base/Dockerfile:GANESHA_VERSION
```

Comment:
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable so it should be possible to have a step like

```bash
perl -ne 'print "$1\n" if /^ARG (GANESHA_VERSION=.+)$/' deploy/base/Dockerfile >> $GITHUB_ENV
```

so that you can just refer subsequently to `${{ GANESHA_VERSION }}` (untested).

Reply:
> so that you can just refer subsequently to ${{ GANESHA_VERSION }} (untested).

using it as `${{ env.GANESHA_VERSION }}`

this action should push image to my repo's ghcr registry on successful completion.
https://github.com/revant/nfs-ganesha-server-and-external-provisioner/actions/runs/4344095874/jobs/7586962446